python/pyspark/worker.py

```diff
@@ -253,7 +253,7 @@ def read_udfs(pickleSer, infile, eval_type):
 
         # NOTE: if timezone is set here, that implies respectSessionTimeZone is True
         timezone = runner_conf.get("spark.sql.session.timeZone", None)
-        ser = ArrowStreamPandasSerializer(timezone)
+        ser = ArrowStreamPandasSerializer(timezone, runner_conf)
```

Comment:
I think it's slightly better to parse the `runner_conf` for the needed options here and pass in the required flags like we do with `timezone`, instead of passing around the `runner_conf`

Reply:
Ok. I passed in the required flag now.